src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -45,24 +52,32 @@
      * @return the entire user of the selected id
      */
     @GetMapping("/{id}")
-    public Optional<Users> findById(@PathVariable Long id) {
-        Users selectedUser;
-        if (userRepository.findById(id).isPresent()) {
-            selectedUser = userRepository.findById(id).get();
-            return Optional.of(selectedUser);
-        } else {
-            return Optional.empty();
-        }
+    @ResponseStatus(HttpStatus.OK)
+    public User findById(@PathVariable Long id) {
+        return userRepository.findById(id)
+                .orElseThrow(UserNotFoundException::new);
+    }
+
+    /**
+     * @param userUsername indicates the username to be searched
+     *
+     * @return the entire user of the selected username
+     */
+    @GetMapping("/getUser/{userUsername}")
+    @ResponseStatus(HttpStatus.OK)
+    public User findById(@PathVariable String userUsername) {
+        return userRepository.findByUserUsername(userUsername)
+                .orElseThrow(UserNotFoundException::new);
```

Comment:
This resource should use a **request param** instead of path variable. Remember
```suggestion
    @GetMapping
    @ResponseStatus(HttpStatus.OK)
    public User findByUsername(@RequestParam String username) {
        return userRepository.findByUserUsername(username)
                .orElseThrow(UserNotFoundException::new);
```